src/libraries/System.Resources.ResourceManager/tests/ResourceManagerTests.cs

```diff
@@ -204,8 +204,19 @@ public static void UsingResourceSet()
         [Fact]
         public static void BaseName()
         {
-            var manager = new ResourceManager("System.Resources.Tests.Resources.TestResx", typeof(ResourceManagerTests).GetTypeInfo().Assembly);
+            var manager = new ResourceManager("System.Resources.Tests.Resources.TestResx", typeof(ResourceManagerTests).Assembly);
+            Assert.Equal("System.Resources.Tests.Resources.TestResx", manager.BaseName);
+
+            manager = new ResourceManager(typeof(System.Resources.Tests.Resources.TestResx));
             Assert.Equal("System.Resources.Tests.Resources.TestResx", manager.BaseName);
+
+            Type typeWithoutNamespace = new { }.GetType();
+            Assert.Null(typeWithoutNamespace.Namespace);
+            manager = new ResourceManager(typeWithoutNamespace);
+            Assert.Equal(typeWithoutNamespace.Name, manager.BaseName);
+
+            manager = new ResourceManager(typeof(List<string>));
+            Assert.Equal("System.Collections.Generic.List`1", manager.BaseName);
```

Comment:
@joperezr This result is kind of interesting. We might expect `String` to be in there somewhere but with the current and previous logic for computing type names the generic parameters don't matter.

Reply:
Is this really a valid case to test? When a generic type could be used?

Reply:
I feel like it is equally valid as any other "irregular" test case (such as testing a type without a namespace). `ResourceManager`'s `Type` constructor allows any type and just looks for resources with the same full name in that type's assembly; AFAIK there are no limitations on what resource files can be named and you don't have to use the built-in generator.

That said, if you don't want this test I'm happy to remove it.

Reply:
As you know the base name is a physical file name without the extension. Even if technically the argument passed could be a generic type and the resource file could be named "System.Collections.Generic.List`1", I don't think it is practical. I think it is uncovered edge case scenario from the beginning and doesn't feel like it's important scenario need to be cover now, if need to cover it should behave same as how it was working before instead of we making new rule for this now, what you think?

Reply:
I think the current test is match with the old behavior, so resolving the comment